mne/preprocessing/maxwell.py

```diff
@@ -102,13 +112,37 @@ def maxwell_filter(raw, origin=(0, 0, 40), int_order=8, ext_order=3,
     pS_tot = pinv(S_tot, cond=1e-15)
     # Compute multipolar moments of (magnetometer scaled) data (Eq. 37)
     mm = np.dot(pS_tot, data * coil_scale[:, np.newaxis])
+
     # Reconstruct data from internal space (Eq. 38)
-    recon = np.dot(S_in, mm[:S_in.shape[1], :])
+    recon = np.dot(S_in, mm[:S_in.shape[1], :]) / coil_scale[:, np.newaxis]
+
+    if st_dur is not None:
+        # Reconstruct data from external space and compute residual
+        recon_out = np.dot(S_out, mm[S_in.shape[1]:, :]) /\
+            coil_scale[:, np.newaxis]
+        resid = data - (recon + recon_out)
+
+        # Generate time points to break up data in to windows
+        t_starts = raw.time_as_index(np.arange(times[0], times[-1], st_dur))
+        t_ends = np.arange(times[0] + st_dur, times[-1], st_dur)
+        # TODO: Decide how to handle last window if it's shorter length
```

Comment:
It would be good to know the rationale behind the 500 -- if it's related to e.g. some multiple of the number of bases, then we could actually codify it instead of using a fixed value.